src/random.cr

```diff
@@ -90,6 +90,20 @@ module Random
     end
   end
 
+  # Returns a random `Float64` which is greater than or equal to 0 and less than *max*.
+  #
+  # ```
+  # Random.new.rand(3.5)   # => 2.88938
+  # Random.new.rand(10.725) # => 7.70147
+  # ```
+  def rand(max : Float) : Float64
+    if max > 0
+      next_u32 * (1 / (UInt32::MAX.to_f64 + 1)) * max
```

Comment:
Good question. For [https://github.com/crystal-lang/crystal/issues/2166#issuecomment-185903735](this). `next_float` return Float in 0..1, including 1. But in `rand(max)` we are needing for 0...max, excluding max. I think, `+ 1` after ` / ` give us a small different, nearly elementary step for Float.